tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/ProcessStart/EnvironmentVariablesAnalyzer.cs

```diff
@@ -0,0 +1,50 @@
+// <copyright file="EnvironmentVariablesAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Process
+{
+    internal static class EnvironmentVariablesAnalyzer
+    {
+        private static string[] allowedEnvVariables = new string[] { "LD_PRELOAD", "LD_LIBRARY_PATH", "PATH" };
+
+        private static bool IsAllowedVariable(string text)
+        {
+            foreach (var allowedVariable in allowedEnvVariables)
+            {
+                if (string.Equals(allowedVariable, text, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        internal static string ScrubbingEnvVariables(IDictionary<string, string> envVariables)
+        {
+            if (envVariables != null)
+            {
+                var variableLine = string.Empty;
+
+                foreach (var variable in envVariables)
+                {
+                    if (IsAllowedVariable(variable.Key))
+                    {
+                        variableLine += variable.Key + "=" + variable.Value + Environment.NewLine;
```

Comment:
Maybe preferable to use a `StringBuilder` here instead of lots of concatenation? Maybe not too bad given it's currently only 3 env vars at most

Reply:
I was thinking about this... Since there are not many concatenations, I was not sure... Anyway, I have changed to StringBuilder, it will probably give us better performance.

Reply:
Also, FYI, there is a `StringBuilderCache` you can use to re-use the `StringBuilder` instance, which may be worth considering 🙂